components/genome-browser-facets.js

```diff
@@ -0,0 +1,536 @@
+      ) : null}
+      <div className="overflow-y-auto h-80">
+        <div className="grid">
+          {Object.keys(facetArray).map((d) => {
+            if (d === "") {
+              // eslint-disable-next-line react/jsx-key
+              return <hr />;
+            }
+            return (
+              <FacetButton
+                selectedFacets={selectedFacets}
+                buttonLabel={`${d} (${facetArray[d]})`}
+                buttonName={d}
+                facetLabel={facetName}
+                addGenomeFilter={addGenomeFilter}
+                key={d}
+              />
+            );
+          })}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+Facet.propTypes = {
+  typeahead: PropTypes.bool.isRequired,
+  facetTitle: PropTypes.string.isRequired,
+  facetName: PropTypes.string.isRequired,
+  facetArray: PropTypes.object.isRequired,
+  addGenomeFilter: PropTypes.func.isRequired,
+  selectedFacets: PropTypes.array.isRequired,
+  handleSearch: PropTypes.func,
+  unsanitizedSearchTerm: PropTypes.string,
+};
+
+class FacetButton extends React.Component {
+  constructor() {
+    super();
+    this._addGenomeFilter = this._addGenomeFilter.bind(this);
+  }
+
+  _addGenomeFilter() {
+    this.props.addGenomeFilter(this.props.buttonName, this.props.facetLabel);
+  }
+
+  render() {
+    const { buttonLabel, buttonName, selectedFacets, facetLabel } = this.props;
+    const isDisabled =
+      buttonLabel.indexOf("(0)") > -1 &&
+      !selectedFacets.includes(`${buttonName}AND${facetLabel}`);
+    const isSelected = selectedFacets.includes(`${buttonName}AND${facetLabel}`);
+    const textColor = isDisabled ? "text-gray-400" : "hover:bg-gray-200";
+    const borderStyle = isSelected ? "border-solid border-2 border-brand" : "";
+    return (
+      <button
+        className={`text-left ${textColor} ${borderStyle}`}
+        onClick={this._addGenomeFilter}
+        disabled={isDisabled}
+      >
+        {buttonLabel}
+      </button>
+    );
+  }
+}
+
+FacetButton.propTypes = {
+  buttonLabel: PropTypes.string.isRequired,
+  buttonName: PropTypes.string.isRequired,
+  facetLabel: PropTypes.string.isRequired,
+  selectedFacets: PropTypes.array.isRequired,
+  addGenomeFilter: PropTypes.func.isRequired,
+};
+
+class FilterButton extends React.Component {
+  constructor() {
+    super();
+    this._addGenomeFilter = this._addGenomeFilter.bind(this);
+  }
+
+  _addGenomeFilter() {
+    this.props.addGenomeFilter(this.props.buttonLabel, this.props.facetLabel);
+  }
+
+  render() {
+    return (
+      <button
+        className="flex"
+        onClick={this._addGenomeFilter}
+        key={this.props.facetKey}
+      >
+        <XCircleIcon className="h-5 w-5" />
+        {this.props.buttonLabel}
+      </button>
+    );
+  }
+}
+
+FilterButton.propTypes = {
+  buttonLabel: PropTypes.string.isRequired,
+  facetLabel: PropTypes.string.isRequired,
+  facetKey: PropTypes.string.isRequired,
+  addGenomeFilter: PropTypes.func.isRequired,
+};
```

Comment:
These previous two class components could easily be converted to functional components if you ever wanted to.